internal/controller/streams.go

```diff
@@ -196,12 +193,38 @@ func (sm *streamManager) StreamResources(ctx context.Context, logger *zap.Logger
 		logger.Error("Failed to send cluster metadata", zap.Error(err))
 		return err
 	}
+
+	resourceVersions := make(map[string]string)
 	for resource, apiGroup := range resourceAPIGroupMap {
-		allResourcesSnapshotted.Add(1)
-		go resourceLister.DynamicListAndWatchResources(ctx, cancel, resource, apiGroup, &allResourcesSnapshotted, &snapshotCompleted)
+		// Add jitter between list resources
+		jitter := time.Duration(rand.Intn(1000)+1000) * time.Millisecond // 1-2 s
+		time.Sleep(jitter)
+		resourceVersion, err := resourceLister.ListK8sResources(ctx, resource, apiGroup)
+		if err != nil {
+			logger.Error("Unable to list resource", zap.String("resource", resource), zap.Error(err))
+			return err
+		}
+		resourceVersions[resource] = resourceVersion
 	}
-	allResourcesSnapshotted.Wait()
+
 	err = sm.sendResourceSnapshotComplete(logger)
+	logger.Info("cloud-operator has succesfully ingested and sent a snapshot of k8s cluster")
+
+	for resource, apiGroup := range resourceAPIGroupMap {
+		version, ok := resourceVersions[resource]
+		if !ok {
+			logger.Warn("Skipping watch: missing resourceVersion", zap.String("resource", resource))
+			return err
+		}
+
+		go func(resource, apiGroup, version string) {
+			// Add jitter between watcher startups
+			jitter := time.Duration(rand.Intn(1000)+1000) * time.Millisecond // 1-2 s
+			time.Sleep(jitter)
```

Comment:
Same as above.
```suggestion
```
If you are worried about doing too many concurrent requests, you could use other mechanisms to serialize the watcher creations that would be better than sleeping for random durations.